### PR TITLE
Resolve #27 by parsing properties

### LINKED ISSUE #27: Replace hard coded styles in TableBuilder._parse_properties by reference to properties
Read in the currently hard coded and thus redundantly present properties from the according file `kursdaten_prop.properties` into the attributes of *TableBuilder* in the method `_parse_properties`. Maybe do something like:
```
settings = open(properties).read().split("\n")
settings_dict = dict()
...
properties = self._prop
```

### PR DESCRIPTION
**Pull request by BjoernLudwig**
This will close #27 and more. Changes we made according to related issues:

- [x] we introduced a `properties.py` and according tests to provide all required inputs at one spot
- [x] we will resolve #35 pep8 conform naming conventions (BREAKING CHANGE)

### CODE REVIEW

**Comment by BjoernLudwigPTB** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- test/test_data/test_properties.py  14
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128&bid=17002520)

**Comment by BjoernLudwigPTB** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- test/test_properties.py  14
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128&bid=17002520)

**Comment by BjoernLudwigPTB** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- test/test_properties.py  14
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128&bid=17002520)

**Comment by BjoernLudwigPTB** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ pyxml2pdf/Main.py  7
         

Complexity increasing per file
==============================
- test/test_properties.py  14
         

Complexity decreasing per file
==============================
+ pyxml2pdf/Main.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128&bid=17002520)

**Comment by BjoernLudwigPTB** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ pyxml2pdf/Main.py  7
         

Complexity increasing per file
==============================
- test/test_properties.py  14
         

Complexity decreasing per file
==============================
+ pyxml2pdf/Main.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128&bid=17002520)

**Comment by BjoernLudwigPTB** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ pyxml2pdf/Main.py  7
         

Complexity increasing per file
==============================
- test/test_properties.py  14
         

Complexity decreasing per file
==============================
+ pyxml2pdf/Main.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128&bid=17002520)

File: test/test_Main.py [hunk 1]
```diff
@@ -24,5 +26,22 @@ def test_init_main():
...
 @pytest.mark.online
 def test_input():
-    with pytest.raises(Exception):
+    with pytest.raises(RuntimeError):
         Main.main()
+
+
+def test_main_call():
+    with pytest.raises(subprocess.CalledProcessError):
+        subprocess.run(["python3", "-m", "pyxml2pdf.Main"], check=True)
```
Comment (BjoernLudwigPTB) [inline test/test_Main.py]: Codacy Issue found: [Starting a process with a partial executable path](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128)

**Comment by BjoernLudwigPTB** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Coverage increased per file
===========================
+ pyxml2pdf/Main.py  31
         

Complexity increasing per file
==============================
- test/test_properties.py  14
- test/test_Main.py  1
         

Complexity decreasing per file
==============================
+ pyxml2pdf/Main.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128&bid=17002520)

File: test/test_Main.py [hunk 2]
```diff
@@ -24,5 +26,22 @@ def test_init_main():
...
+    with pytest.raises(subprocess.CalledProcessError):
+        subprocess.run(["python3", "-m", "pyxml2pdf.Main", "test.xml"], check=True)
+    with pytest.raises(subprocess.CalledProcessError):
+        subprocess.run(["python3", "-m", "pyxml2pdf.Main", "test.test"], check=True)
+    with pytest.raises(subprocess.CalledProcessError):
+        subprocess.run(
+            ["python3", "-m", "pyxml2pdf.Main", "test.xml", "test.test"], check=True
+        )
+    with pytest.raises(subprocess.CalledProcessError):
+        subprocess.run(
```
Comment (BjoernLudwigPTB) [inline test/test_Main.py]: Codacy Issue found: [Starting a process with a partial executable path](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128)

File: test/test_Main.py [hunk 3]
```diff
@@ -24,5 +26,22 @@ def test_init_main():
...
-    with pytest.raises(Exception):
+    with pytest.raises(RuntimeError):
         Main.main()
+
+
+def test_main_call():
+    with pytest.raises(subprocess.CalledProcessError):
+        subprocess.run(["python3", "-m", "pyxml2pdf.Main"], check=True)
+    with pytest.raises(subprocess.CalledProcessError):
+        subprocess.run(["python3", "-m", "pyxml2pdf.Main", "test.xml"], check=True)
```
Comment (BjoernLudwigPTB) [inline test/test_Main.py]: Codacy Issue found: [Starting a process with a partial executable path](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128)

File: test/test_Main.py (hunk 2 above)
Comment (BjoernLudwigPTB) [inline test/test_Main.py]: Codacy Issue found: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128)

File: test/test_Main.py (hunk 1 above)
Comment (BjoernLudwigPTB) [inline test/test_Main.py]: Codacy Issue found: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128)

File: test/test_Main.py [hunk 4]
```diff
@@ -24,5 +26,22 @@ def test_init_main():
...
+    with pytest.raises(subprocess.CalledProcessError):
+        subprocess.run(["python3", "-m", "pyxml2pdf.Main", "test.test"], check=True)
```
Comment (BjoernLudwigPTB) [inline test/test_Main.py]: Codacy Issue found: [Starting a process with a partial executable path](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128)

File: test/test_Main.py (hunk 3 above)
Comment (BjoernLudwigPTB) [inline test/test_Main.py]: Codacy Issue found: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128)

File: test/test_Main.py [hunk 5]
```diff
@@ -1,3 +1,5 @@
+import subprocess
```
Comment (BjoernLudwigPTB) [inline test/test_Main.py]: Codacy Issue found: [Consider possible security implications associated with subprocess module.](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128)

File: test/test_Main.py (hunk 4 above)
Comment (BjoernLudwigPTB) [inline test/test_Main.py]: Codacy Issue found: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128)

**Comment by BjoernLudwigPTB** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Coverage increased per file
===========================
+ pyxml2pdf/Main.py  31
         

Complexity increasing per file
==============================
- test/test_properties.py  14
- test/test_Main.py  1
         

Complexity decreasing per file
==============================
+ pyxml2pdf/Main.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128&bid=17002520)

File: test/test_Main.py [hunk 6]
```diff
@@ -22,7 +24,30 @@ def test_init_main():
...
-    with pytest.raises(Exception):
+    with pytest.raises(RuntimeError):
         Main.main()
+
+
+def test_main_call():
+    with pytest.raises(subprocess.CalledProcessError):
+        subprocess.run(["python3", "-m", "pyxml2pdf.Main"], check=True)
+    with pytest.raises(subprocess.CalledProcessError):
+        subprocess.run(["python3", "-m", "pyxml2pdf.Main", "test.xml"], check=True)
```
Comment (BjoernLudwigPTB) [inline test/test_Main.py]: Codacy Issue found: [Starting a process with a partial executable path](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128)

File: test/test_Main.py [hunk 7]
```diff
@@ -22,7 +24,30 @@ def test_init_main():
...
+    with pytest.raises(subprocess.CalledProcessError):
+        subprocess.run(["python3", "-m", "pyxml2pdf.Main", "test.test"], check=True)
+    with pytest.raises(subprocess.CalledProcessError):
+        subprocess.run(
+            ["python3", "-m", "pyxml2pdf.Main", "test.xml", "test.test"], check=True
+        )
+    with pytest.raises(subprocess.CalledProcessError):
+        subprocess.run(
```
Comment (BjoernLudwigPTB) [inline test/test_Main.py]: Codacy Issue found: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128)

File: test/test_Main.py [hunk 8]
```diff
@@ -22,7 +24,30 @@ def test_init_main():
...
 @pytest.mark.online
 def test_input():
-    with pytest.raises(Exception):
+    with pytest.raises(RuntimeError):
         Main.main()
+
+
+def test_main_call():
+    with pytest.raises(subprocess.CalledProcessError):
+        subprocess.run(["python3", "-m", "pyxml2pdf.Main"], check=True)
```
Comment (BjoernLudwigPTB) [inline test/test_Main.py]: Codacy Issue found: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128)

File: test/test_Main.py (hunk 7 above)
Comment (BjoernLudwigPTB) [inline test/test_Main.py]: Codacy Issue found: [Starting a process with a partial executable path](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128)

File: test/test_Main.py (hunk 5 above)
Comment (BjoernLudwigPTB) [inline test/test_Main.py]: Codacy Issue found: [Consider possible security implications associated with subprocess module.](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128)

File: test/test_Main.py [hunk 9]
```diff
@@ -22,7 +24,30 @@ def test_init_main():
...
+    with pytest.raises(subprocess.CalledProcessError):
+        subprocess.run(["python3", "-m", "pyxml2pdf.Main", "test.xml"], check=True)
+    with pytest.raises(subprocess.CalledProcessError):
+        subprocess.run(["python3", "-m", "pyxml2pdf.Main", "test.test"], check=True)
```
Comment (BjoernLudwigPTB) [inline test/test_Main.py]: Codacy Issue found: [Starting a process with a partial executable path](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128)

Comment (BjoernLudwigPTB) [inline test/test_Main.py]: Codacy Issue found: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128)

File: test/test_Main.py (hunk 6 above)
Comment (BjoernLudwigPTB) [inline test/test_Main.py]: Codacy Issue found: [subprocess call - check for execution of untrusted input.](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128)

File: test/test_Main.py (hunk 8 above)
Comment (BjoernLudwigPTB) [inline test/test_Main.py]: Codacy Issue found: [Starting a process with a partial executable path](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128)

**Comment by BjoernLudwig** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ pyxml2pdf/Main.py  7
         

Complexity increasing per file
==============================
- test/test_properties.py  14
         

Complexity decreasing per file
==============================
+ pyxml2pdf/Main.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/blus_projects/pyxml2pdf/pullRequest?prid=5181128&bid=17002520)

**Pull request merged**